README.markdown

```diff
@@ -529,30 +836,34 @@ If the resource already exists, but does not match the specified parameters, thi
 
 An array of resources can also be passed in, and each will be created with the type and parameters specified if it doesn't already exist.
 
-  `ensure_resource('user', ['dan','alex'], {'ensure' => 'present'})`
+`ensure_resource('user', ['dan','alex'], {'ensure' => 'present'})`
 
 *Type*: statement.
 
 #### `ensure_resources`
 
-Takes a resource type, title (only hash), and a hash of attributes that describe the resource(s).
+TODO: I'm not sure I understand this one exactly. Need to clarify and rewrite.
```

Comment:
I'll take a bash describing here, forgive me if it doesn’t help!
So if I want to ensure a resource is exists I can use this function. I pass in the type of the resource I'm looking for, it's title and a hash of attributes that will be checked against the resource, and if the resource isn’t present it will attempt creating it. 
So with the example:
`ensure_resource('user', 'jean', {'ensure' => 'present'})`
I am trying to ensure there is a user resource called jean, and I want to make sure it has attributes 'ensure' => 'present'.
If the resource does not exist it will create it:  user { ‘jean’:
	ensure => present,
}

If the (jean) resource already exists but does not match the specified parameters:
user { ‘jean’:
	ensure => absent,
}
this function will attempt to recreate the resource leading to a duplicate
resource definition error. Rough example: Error: Duplicate declaration: User[jean] is already declared in file /

In the other example it uses a hash of titles - this simply loops through the titles performs the aforementioned functionality on each of them.